terraform/modules/monitoring/monitoring.tf

```diff
@@ -0,0 +1,135 @@
+variable "gcp_project" {
+  type = string
+}
+
+variable "gcp_region" {
+  type = string
+}
+
+resource "kubernetes_namespace" "monitor" {
+  metadata {
+    name = "monitor"
+  }
+}
+
+resource "kubernetes_service" "prometheus" {
+  metadata {
+    name      = "prometheus"
+    namespace = "monitor"
+  }
+  spec {
+    selector = {
+      app = kubernetes_pod.prometheus.metadata.0.labels.app
+    }
+    port {
+      port        = 9090
+      target_port = 9090
+    }
+
+    type = "LoadBalancer"
+  }
+}
+
+resource "kubernetes_pod" "prometheus" {
+  metadata {
+    name      = "prometheus"
+    namespace = "monitor"
+    labels = {
+      app = "Prometheus"
+    }
+  }
+
+  spec {
+    service_account_name = kubernetes_service_account.prometheus_account.metadata[0].name
+    container {
+      image = "prom/prometheus:latest"
+      name  = "prometheus"
+    }
+    volume {
+      name = "config-volume"
+      config_map {
+        name = "prometheus_config"
+      }
+    }
+  }
+}
+
+resource "kubernetes_config_map" "prometheus_config" {
+  metadata {
+    name = "prometheus-config"
+  }
+
+  data = {
+    "prometheus.yml" = file("${path.module}/prometheus.yml")
+  }
+}
+
+resource "kubernetes_cluster_role" "prometheus_cluster_role" {
+  metadata {
+    name = "prometheus-role"
+    #namespace = "monitor"
```

Comment:
Is this role not namespaced? There's a distinction between a `kubernetes_cluster_role` and a [`kubernetes_role`](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/role).

Reply:
Correct, this is meant to be a cluster role (not a role) because it has to be non-namespaced in order to reach into all the namespaces.